addok_psql_store/__init__.py

```diff
@@ -25,7 +25,14 @@ def __init__(self, *args, **kwargs):
     def getconn(self):
         # Use pid as connection id so we can reuse the connection within the
         # same process.
-        return self.pool.getconn(key=os.getpid())
+        conn = self.pool.getconn(key=os.getpid())
+        try:
+            c = conn.cursor()
+        except OperationalError as err:
+            self.pool = pool.SimpleConnectionPool(minconn=8, maxconn=64,
```

Comment:
it will leak all others active connections, and if we call closeall before, it will close eventually active healthy connections,
i think it is better to put back the connection `self.pool.putconn(key=os.getpid())`, it will be discarded regarding the code here (https://github.com/psycopg/psycopg2/blob/master/lib/pool.py) and to try to get another one by calling ourselve to try recursively